borgmatic/commands/completion.py

```diff
@@ -55,3 +65,153 @@ def bash_completion():
             '\ncomplete -o bashdefault -o default -F complete_borgmatic borgmatic',
         )
     )
+
+
+# fish section
+
+
+def has_file_options(action: Action):
+    return action.metavar in (
+        'FILENAME',
+        'PATH',
+    ) or action.dest in ('config_paths',)
+
+
+def has_choice_options(action: Action):
+    return action.choices is not None
+
+
+def has_required_param_options(action: Action):
+    return (
+        action.required is True
+        or action.nargs
+        in (
+            '+',
+            '*',
+        )
+        or '--archive' in action.option_strings
+        or action.metavar in ('PATTERN', 'KEYS', 'N')
+        or (action.type is not None and action.default is None)
+    )
+
+
+def has_exact_options(action: Action):
+    return (
+        has_file_options(action) or has_choice_options(action) or has_required_param_options(action)
+    )
+
+
+def exact_options_completion(action: Action):
+    '''
+    Given an argparse.Action instance, return a completion invocation
+    that forces file completion or options completion, if the action
+    takes such an argument and was the last action on the command line.
```

Comment:
FYI you can wrap at 100 characters in this codebase.

Reply:
Is there a way to automate this here? The formatters don't seem to do it and I don't have a good eye for character count.

Reply:
Depends on your IDE.

 * VSCode: https://stackoverflow.com/questions/43122175/automatically-hard-wrap-lines-at-column-in-vscode
 * vim: https://vi.stackexchange.com/questions/18798/automatic-wraping-to-new-line